R/processes.R

```diff
@@ -344,3 +363,28 @@ create_lagged_eir <- function(variables, solvers, parameters) {
     }
   )
 }
+
+#' @title Hypnozoite decay function
+#' @description
+#' calulates the number of decaying hypnozoites in each individual
+#'
+#' @param variable the hypnozoite variable to update
+#' @param rate the hypnozoite decay rate
+#' @noRd
+create_hypnozoite_decay_process <- function(hypnozoites, gammal, renderer){
+  function(timestep){
+    if(hypnozoites$get_size_of(!0)>0){
+      hyp_bitset <- hypnozoites$get_index_of(0)$not(TRUE)
+      n_hypnozoites <- hypnozoites$get_values(index = hyp_bitset)
+      hypozoite_decay <- vnapply(
+        n_hypnozoites,
+        function(x) rbinom(n = 1, size = x, prob = gammal)
+      )
+      hypnozoites$queue_update(
+        values = n_hypnozoites - hypozoite_decay,
+        index = hyp_bitset
+      )
+      renderer$render('n_decayed', sum(hypozoite_decay), timestep)
+    }
+  }
+}
```

Comment:
Is this slightly more readable?

```suggestion
create_hypnozoite_decay_process <- function(hypnozoites, gammal, renderer){
  function(timestep){
    hyp_bitset <- hypnozoites$get_index_of(0)$not(TRUE)
    if(hyp_bitset$size() > 0){      
      n_hypnozoites <- hypnozoites$get_values(index = hyp_bitset)
      hypozoite_decay <- vnapply(
        n_hypnozoites,
        function(x) rbinom(n = 1, size = x, prob = gammal)
      )
      hypnozoites$queue_update(
        values = n_hypnozoites - hypozoite_decay,
        index = hyp_bitset
      )
      renderer$render('n_decayed', sum(hypozoite_decay), timestep)
    }
  }
}
```

Reply:
I've added this change in the next commit.